geniza/corpus/management/commands/merge_joins.py

```diff
@@ -18,12 +24,28 @@
 from geniza.corpus.models import Document
 
 
+@dataclass
+class MergeGroup:
+    """Group of Documents that are ready to be merged: candidates in `to_merge`
+    will be combined with `primary` using Document.merge_with, creating log
+    entries to document the merge reasoning using `rationale`."""
+
+    primary: int = None  # pgpid
+    to_merge: List[int] = field(default_factory=list)  # list of pgpid
+    rationale: str = ""
+
+    def merge(self):
+        """Execute the merge using `Document.merge_with()`."""
+        Document.objects.get(self.primary).merge_with(self.to_merge, self.rationale)
```

Comment:
i don't think we decided whether the API for this function would accept an iterable of `Document` or an iterable of pgpids for the first param. i think it might be more consistent to do the former, so that the caller is responsible for retrieving all the objects from the db (since it already has to retrieve the primary). if that sounds good to you I can change it here, and add a bit of error handling in the manage command in case we get a `Document.DoesNotExist` from a bad pgpid.

Reply:
From what we discussed, I expected it to be given a list of documents (not ids). I agree that it would be good to have the error handling external to the function